### DISCUSSION TITLE
Error in Calling an RFC Returning a Field and Table with Purchase Orders

### DISCUSSION THREAD

**Discussion by fitzgeraldtorres** [dbosoft/YaNco#321] [Q&A]
I need help calling an RFC that returns a field and a table with purchase order records, but the function throws an error indicating that the type arguments cannot be inferred from the usage. I’m attaching my function and the error code. Thank you in advance for your help.

public async Task<BuyDocumentResponseDto> TraePedidoCompraAsync(BuyDocumentRequestDto request)
{
    var sapSettings = _configuration.GetSection("SapSettings").Get<SapConfigDto>();
    var settings = new Dictionary<string, string>
    {
        {"ashost", sapSettings.AppServerHost},
        {"sysnr", sapSettings.SystemNumber},
        {"client", sapSettings.Client},
        {"user", sapSettings.User},
        {"passwd", sapSettings.Password},
        {"lang", sapSettings.Language}
    };

    var connectionBuilder = new ConnectionBuilder(settings);
    var connFunc = connectionBuilder.Build();

    using (var context = new RfcContext(connFunc))
    {

       var result = await context.CallFunction("Z_RFC_TRAE_PEDIDO_COMPRA_003",
            Input: f =>
            {
                f.SetField("CENTRO", request.Centro);
                f.SetField("MATERIAL", request.Material);
            },
            Output: f => from Order in f.MapTable("PEDIDO", s =>  from Ebeln in s.GetField<string>("EBELN")
                                                    from Ebelp in s.GetField<string>("EBELP")
                                                    from Matnr in s.GetField<string>("MATNR")
                                                    from Menge in s.GetField<string>("MENGE")
                                                    from Meins in s.GetField<string>("MEINS")
                                                    from Lgort in s.GetField<string>("LGORT")
                                                    from Ktmng in s.GetField<string>("KTMNG")
                                                    from Loekz in s.GetField<string>("LOEKZ")
                                                    from Abdat in s.GetField<string>("ABDAT")
                                                    from Labnr in s.GetField<string>("LABNR")
                                                    select new  {Ebeln, Ebelp, Matnr, Menge, Meins, Lgort, Ktmng, Loekz, Abdat, Labnr }
                                                  )
                          from Error in f.GetField<string>("ERROR")
                          select new { Order, Error }
                        )
             .Match(l => Console.WriteLine($"Error: {l.Message}"));

        var response = new BuyDocumentResponseDto
        {
            Error = result.Error,
            Pedido = result.Select(o => new BuyDocument
            {
                Ebeln = o.Ebeln,
                Ebelp = o.Ebelp,
                Matnr = o.Matnr,
                Menge = o.Menge,
                Meins = o.Meins,
                Lgort = o.Lgort,
                Ktmng = o.Ktmng,
                Loekz = o.Loekz,
                Abdat = o.Abdat,
                Labnr = o.Labnr
            }).ToList()
        };

        return response;
    }
}

<img width="1276" alt="Captura de Pantalla 2024-12-13 a la(s) 11 46 06" src="https://github.com/user-attachments/assets/d3ec17f2-7d90-4235-a18f-21bf4310d4e8" />

**Comment by fw2568** [comment]
Your input method does not return a value. See https://github.com/dbosoft/YaNco?tab=readme-ov-file#calling-abap-function-modules for correct usage.
